lib/IRGen/GenDecl.cpp

```diff
@@ -571,7 +571,11 @@ emitGlobalList(IRGenModule &IGM, ArrayRef<llvm::WeakTrackingVH> handles,
                                       init, name);
   var->setSection(section);
   var->setAlignment(llvm::MaybeAlign(alignment.getValue()));
-  disableAddressSanitizer(IGM, var);
+
+  // Do not apply disableAddressSanitizer to @llvm.used and @llvm.compiler.used
+  // because that confuses ThinLTO and crashes when merging these lists.
+  if (name != "llvm.used" && name != "llvm.compiler.used")
```

Comment:
I think it might be better to special case these two inside of `disableAddressSanitizer`.